# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/xf86-video-r128/hotfix-no-vbios.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

Fix X not crashing on Apple PowerMacs w/o mapped VBIOS.

Start to tweak DVI output support, though DDC data does not yet come back?

--- xf86-video-r128-6.13.0/src/r128_output.c.vanilla	2024-11-03 17:36:23.783436402 +0100
+++ xf86-video-r128-6.13.0/src/r128_output.c	2024-11-03 19:57:45.053590548 +0100
@@ -343,10 +344,7 @@
         if (r128_output->type == OUTPUT_VGA) {
             MonType = MT_CRT;
         } else {
-            if ((*MonInfo)->rawData[0x14] & 0x80)
-                MonType = MT_DFP;
-            else
-                MonType = MT_CRT;
+            MonType = MT_DFP;
         }
     }
 
@@ -477,12 +479,13 @@
     /* non-x86 platform */
     if (!info->VBIOS) {
         if (info->isDFP) {
-            /* XXX assume LVDS on mobility chips */
-            otypes[0] = OUTPUT_LVDS;
+            /* TODO: assume LVDS on mobility chips only, DVI otherwise */
+            otypes[0] = OUTPUT_DVI; // LVDS
             otypes[1] = OUTPUT_VGA;
         } else {
             otypes[0] = OUTPUT_VGA;
         }
+	return;
     }
 
     bios_header = R128_BIOS16(0x48);
